src/popup/router/components/TransactionOverview.vue

```diff
@@ -0,0 +1,282 @@
+        <div class="line" />
+      </div>
+      <div v-if="transaction.aens" class="avatar">
+        <AensIcon class="icon" />
+      </div>
+      <div v-else-if="transaction.contractCreate" class="avatar">
+        <ContractIcon class="icon" />
+      </div>
+      <Avatar v-else v-bind="transaction.recipient" />
+    </div>
+
+    <div class="details">
+      <div class="sender">
+        <a
+          :href="connect ? app.host : getExplorerPath(transaction.sender.address)"
+          target="blank"
+          class="name"
+        >
+          <TruncateMid :str="transaction.sender.name || transaction.sender.label" />
+        </a>
+        <CopyAddress :value="transaction.sender.address" />
+      </div>
+      <div class="recipient">
+        <span
+          v-if="transaction.aens || transaction.contractCreate"
+          :class="{ aens: transaction.aens }"
+          class="name"
+        >
+          {{ transaction.recipient.label }}
+        </span>
+        <a
+          v-else
+          :href="getExplorerPath(transaction.recipient.address)"
+          target="blank"
+          class="name"
+        >
+          <TruncateMid :str="transaction.recipient.name || transaction.recipient.label" />
+        </a>
+        <CopyAddress v-if="transaction.recipient.address" :value="transaction.recipient.address" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+import { OBJECT_ID_TX_TYPE } from '@aeternity/aepp-sdk/es/tx/builder/schema';
+import { TX_TYPES, TX_TYPES_MDW } from '../../utils/constants';
+import TriangleRight from '../../../icons/triangle-right.svg?vue-component';
+import ContractIcon from '../../../icons/contract-create.svg?vue-component';
+import AensIcon from '../../../icons/aens.svg?vue-component';
+import Avatar from './Avatar';
+import TruncateMid from './TruncateMid';
+import CopyAddress from './CopyAddress';
+
+export default {
+  name: 'TransactionOverview',
+  components: {
+    Avatar,
+    TruncateMid,
+    CopyAddress,
+    TriangleRight,
+    AensIcon,
+    ContractIcon,
+  },
+  props: {
+    tx: { type: Object },
+    claim: { type: Boolean },
+    event: { type: String },
+    contract: { type: String },
+    connect: { type: Boolean },
+    app: { type: Object },
+  },
+  computed: {
+    ...mapGetters(['account', 'getExplorerPath']),
+    transaction() {
+      if (this.connect)
+        return {
+          sender: {
+            address: this.app.host,
+            name: this.app.name,
+          },
+          recipient: this.currentAccount,
+          title: this.$t('pages.connectConfirm.connecting'),
+          desc: '...',
+        };
+      if (this.claim)
+        return {
+          sender: {
+            address: this.contract,
+            label: this.$t('transaction.overview.contract'),
+          },
+          recipient: this.currentAccount,
+          title: this.$t('transaction.overview.receiveTips'),
+          desc: this.event,
+        };
```

Comment:
I would handle this as a general contract call transaction, this approach doesn't care about third-party developers, show the contract's method name instead.

Reply:
Tip claim transactions are fetched from https://raendom-backend.z52da5wt.xyz/cache/events/?address=...&event=TipWithdrawn and have different fields from a general contract call transaction:

```
{
    "event": "TipWithdrawn",
    "caller": "ak_HryHrgaVUSxezLQBfNENhptKeXk6tpg4kjWBk88nsU5RJvhNk",
    "nonce": 11408,
    "height": 380886,
    "hash": "th_KCGPBwhc7mKo2rnuQfSpAKaFsh6zoyMmnLW93BiUP5QSmmHFf",
    "time": 1612287712669,
    "contract": "ct_2AfnEfCSZCTEkxL5Yoi4Yfq6fF7YapHRaFKDJK3THMXMBspp5z",
    "address": "ak_DUFY68J5QvRTDUEoAKSVXgYGovV86dyboNk8RVq3gvwQ1fQ5y",
    "amount": "120000000000000000",
    "url": "https://github.com/imollov",
    "tokenContract": null
  }
```

Reply:
Later middleware will provide a single list of transactions that have changed user's balance, for we can mimic it by something like:
```js
    let txs = await Promise.all([
      state.middleware.getTxByAccount(address, limit, page).then(({ data }) => data),
      dispatch('fetchPendingTransactions'),
      fetchJson(
        `${getters.activeNetwork.backendUrl}/cache/events/?address=${address}&event=TipWithdrawn${
          recent ? `&limit=${limit}` : ``
        }`,
      )
        .then((response) =>
          response.map(({ hash }) => state.middleware.getTxByHash(hash)),
        )
        .catch(() => []),
    ]);
```
[here](https://github.com/aeternity/superhero-wallet/blob/28cdf10f00df58ecc7efe7bd8593b6e9939e4ffe/src/store/actions.js#L40-L57)